### ISSUE TITLE
[Sync] ClientReset, using the backup Realm file is not possible

### ISSUE THREAD

**Issue by nhachicha** [realm/realm-java#4997]
Following the concerns from this post https://forums.realm.io/t/how-can-i-test-my-apps-error-handling/393/7
it seems we don't expose a way to reuse the backup Realm file (after a ClientReset). The current API returns the full path name, making it impossible to build a new `SyncConfiguration` that targets the saved file, so the user can copy data from it, making the saved Realm nearly unusable.

**Comment by Zhuinden** [comment]
That's interesting, the change log says that this is relevant to #4080

**Comment by sipersso** [comment]
@nhachicha thank you for noticing my forum post. Right now, I have just made an additional copy of the Realm using Realm.writeToFile() before executing the reset, but I am still struggling to get the data over and I haven't found any examples on how to execute a client reset properly.

If there is anything I can do to help, let me know.

**Comment by Zhuinden** [comment]
It's a bit odd though as I would think https://realm.io/docs/java/latest/api/io/realm/ClientResetRequiredError.html should provide the necessary data.

``` java
error.executeClientReset();

// Validate that files have been moved
assertFalse(error.getOriginalFile().exists());
assertTrue(error.getBackupFile().exists());
```

**Comment by sipersso** [comment]
@Zhuinden agreed! And I think a little more clarification on how to actually handle the ClientResets in a proper way would be great.

**Comment by sipersso** [comment]
@nhachicha any info on why this is blocked? Is there any more info on how to handle ClientResets properly?

**Comment by nhachicha** [comment]
The implementation should happen at the [Object Store](https://github.com/realm/realm-object-store) level. The currently exported file has a history that needs a `SyncConfiguration` in order to be opened. We're experimenting a workaround at the ObjectStore level in order to allow the user to open the Realm without specifying SyncConfig. 
stay tuned!

**Comment by cmelchior** [comment]
Implemented in #4759. Closing
